release_docs/RELEASE.txt

```diff
@@ -207,6 +207,21 @@ Bug Fixes since HDF5-1.13.3 release
 ===================================
     Library
     -------
+    - Fixed an issue with collective metadata writes of global heap data
```

Comment:
Shouldn't this statement be under "Bug Fixes since HDF5-1.14.0 release", instead of 1.13.3?